apps/web/src/services/onboard/ledger-module.ts

```diff
@@ -159,12 +159,31 @@ export function ledgerModule(): WalletInit {
                 value: txParams.value ? BigInt(txParams.value) : null,
               })
 
-              transaction.signature = await ledgerSdk.signTransaction(
-                getAssertedDerivationPath(),
-                hexaStringToBuffer(transaction.unsignedSerialized)!,
-              )
+              // Calculate hash and show comparison dialog before signing
+              const { keccak256 } = await import('ethers')
+              const txHash = keccak256(transaction.unsignedSerialized)
 
-              return transaction.serialized
+              const { showLedgerHashComparison, hideLedgerHashComparison } = await import(
+                '@/components/common/LedgerHashComparison'
+              )
```

Comment:
does this need to be inline. 

The show and hide functions are utility functions/business logic - they have nothing to do with the LedgerHashComparison component. I mean they are not part of the component itself. They should live in a separate file than the component UI.